envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -210,18 +211,61 @@ class EnvoyClustersFactory(
         return clustersForGroup
     }
 
-    private fun createClusterForGroup(dependencySettings: DependencySettings, cluster: Cluster?): Cluster? {
+    private fun createClusterForGroup(
+        dependencySettings: DependencySettings,
+        cluster: Cluster?,
+        defaultDependencySettings: DependencySettings? = null
+    ): Cluster? {
         return cluster?.let {
             val idleTimeoutPolicy =
-                dependencySettings.timeoutPolicy.connectionIdleTimeout ?: cluster.commonHttpProtocolOptions.idleTimeout
+                dependencySettings.timeoutPolicy.connectionIdleTimeout
+                    ?: defaultDependencySettings?.timeoutPolicy?.connectionIdleTimeout
+                    ?: cluster.commonHttpProtocolOptions.idleTimeout
```

Comment:
seems to be depercated

Reply:
https://github.com/allegro/envoy-control/pull/302/files/ac7e60c5c2587272caeeceffc3eca2ff979af705#r827740475